### ISSUE TITLE
Ingame connection issues

### ISSUE THREAD

**Issue by Sebola3461** [ppy/osu-stable-issues#868]
for a long time, I'm having connection issues (Bancho server only). this is preventing me from playing multiplayer, today I lost a tournament because of this, so opening this issue. the only thing my logs say is: "WARNING: ThreadPool is saturated!"

[session.log](https://github.com/ppy/osu-stable-issues/files/7444581/session.log)
[runtime.log](https://github.com/ppy/osu-stable-issues/files/7444582/runtime.log)
[network.log](https://github.com/ppy/osu-stable-issues/files/7444584/network.log)

**Comment by peppy** [comment]
Hard to say anything as the logs are empty. Which means there have been no network requests since you started the game. Please provide logs after you encounter the issue (before restarting the game).

**Comment by Sinamer** [comment]
I've been having the same issue, sent and email to accounts@ppy.sh half a year ago with no answer so far:
![image](https://user-images.githubusercontent.com/53346335/140645729-ec8ca3b9-3f6f-40df-830f-4557abbd0e9b.png)

The problem with logs is that you can't connect to multiplayer on Cutting Edge (Experimental) release without being osu! supporter, so all the information in network.log is just "WARNING: Thread Pool is saturated!"

**Comment by peppy** [comment]
@Sinamer can you confirm you're still experiencing the same issues? have you been able to test on a different physical network connection since then?

**Comment by Sinamer** [comment]
Yes, I am still experiencing the same issues on my home network connection. As for a different physical network, tried it just now and it has no connection issues (the network.log is also never created, since osu! didn't run into any issues)

**Comment by Sebola3461** [comment]
i forgot to say, its the same error

> I've been having the same issue, sent and email to [accounts@ppy.sh](mailto:accounts@ppy.sh) half a year ago with no answer so far: ![image](https://user-images.githubusercontent.com/53346335/140645729-ec8ca3b9-3f6f-40df-830f-4557abbd0e9b.png)
> 
> The problem with logs is that you can't connect to multiplayer on Cutting Edge (Experimental) release without being osu! supporter, so all the information in network.log is just "WARNING: Thread Pool is saturated!"

**Comment by peppy** [comment]
Need more logs coming from cutting edge installations to give any answers here. stable stream doesn't give enough log output to make any calls.

**Comment by Sinamer** [comment]
Unfortunately, I can't access multiplayer in Cutting Edge to get proper log output:
![screenshot097](https://user-images.githubusercontent.com/53346335/140924879-d9e0304c-2fa2-4a9b-85e8-2bf9301a1a94.png)

However, after some more testing I found out that ThreadPool warnings are logged only when you open the lobby screen (see screenshot below), which makes me think those warnings are not related to connection loss in multiplayer. Also, I have confirmed the warnings appear regardless of a physical network I'm connected to (I said it made the warning go away previously, but I was wrong after double-checking).
![screenshot098](https://user-images.githubusercontent.com/53346335/140926457-c812a4b6-5b90-495b-b631-f0d9d6c18313.png)

As for the connection issues, it seems to vary, sometimes being completely normal and sometimes being really bad. At this point I believe it's wireless hardware or an issue with ISP.
Would like to be proven wrong though, if the ThreadPool warning is resolved and it fixes the occasional connection loss it would be great.

**Comment by Sebola3461** [comment]
More people with the same issue
![image](https://user-images.githubusercontent.com/62405013/149644714-75b799cb-d0c2-464c-a913-e6b6d36d165a.png)
